hicommonwealth/commonwealth
# Add Amino support for CommunityPoolSpendProposal

## Link to Issue
Closes: #4731 

## Description of Changes
- Adds Ledger (amino) support for all supported Cosmos onchain transactions except:
    -  `MsgSubmitProposal` with `CommunityPoolSpendProposal ` in **v1** (note that this does support it in `v1beta1`)
    - The automated test for this is skipped and marked for new issue...
- Bumps cosmJs to 0.31.0, and makes associated migration changes


## "How We Fixed It"

Added gov/aminomessages.ts to amino registry when creating signingClient

## Test Plan
- Note: I (or anybody) can send tokens to whoever wants to QA with a Ledger.
    - Recommended for QA: 
        - if running Redis: `redis-cli` > `FLUSHALL` (this will clear devnet sandbox params which change with this PR)
        - In Keplr, Settings > General > Manage Non-native Chains > Delete Cosmos Devnet chains (we will re-add them) 
- Amino tests added to devnet suite (create both types of proposals, vote).
- CA (click around) tested on local:
  - Using a Ledger device with staked tokens, 
  - create Text Proposal and Community Spend Proposal on /csdk-beta, and vote
  - create Text Proposal and vote on /csdk (spend prop for v1 is a follow-up, see below)
  

## Deployment Plan

1. 

## Other Considerations

- #4821 for v1 community spend
